File: python/python_bandwidth_bxml/verbs/speak_sentence.py
"""
speak_sentence.py

Representation of Bandwidth's speak sentence BXML verb

@author Jacob Mulford
@copyright Bandwidth INC
"""

from lxml import etree

from .base_verb import AbstractBxmlVerb

SPEAK_SENTENCE_TAG = "SpeakSentence"


class SpeakSentence(AbstractBxmlVerb):

    def __init__(self, sentence=None, voice=None, locale=None):
        """
        Initializes the SpeakSentence class with the following parameters

        :param str sentence: The sentence to speak 
        :param str voice: The voice to speak the sentence
        :param str locale: The locale of the voice
        """
        self.sentence = sentence
        self.voice = voice
        self.locale = locale

    def to_xml(self):
        root = etree.Element(SPEAK_SENTENCE_TAG)
        if self.sentence is not None:
            root.text = self.sentence
        if self.voice is not None:
            root.set("voice", self.voice)
        if self.locale is not None:
            root.set("locale", self.locale)
        return etree.tostring(root).decode()

File: nodejs/lib/verbs/speakSentence.js
/**
 * speakSentence.js
 *
 * BXML generator for the SpeakSentence verb
 *
 * @author Jacob Mulford
 * @copyright Bandwidth INC
 */

var BaseVerb = require("./baseVerb");

class SpeakSentence extends BaseVerb {

    constructor() {
        super();
        this.sentence = null;
        this.voice = null;
        this.locale = null;
        this.gender = null;
    }

    /**
     * Sets the sentence attribute
     *
     * @param {String} sentence The value to set sentence to
     */
    setSentence(sentence) {
        this.sentence = sentence;
    }

    /**
     * Sets the voice attribute
     *
     * @param {String} voice The value to set voice to
     */
    setVoice(voice) {
        this.voice = voice;
    }

    /**
     * Sets the locale attribute
     *
     * @param {String} locale The value to set locale to
     */
    setLocale(locale) {
        this.locale = locale;
    }

    /**
     * Sets the gender attribute
     *
     * @param {String} gender The value to set gender to
     */
    setGender(gender) {
        this.gender = gender;
    }

    addXml(xmlDocument) {
        var attributes = {};

        if (this.voice !== null) {
            attributes["voice"] = this.voice;
        }

        if (this.locale !== null) {
            attributes["locale"] = this.locale;
        }

        if (this.gender !== null) {
            attributes["gender"] = this.gender;
        }

        xmlDocument.ele("SpeakSentence", attributes, this.sentence);
        return xmlDocument;
    }
}

module.exports = SpeakSentence;
